app/src/main/java/org/simple/clinic/registration/facility/RegistrationFacilitySelectionEffectHandler.kt

```diff
@@ -40,4 +43,16 @@ class RegistrationFacilitySelectionEffectHandler @AssistedInject constructor(
           .map(::LocationFetched)
     }
   }
+
+  private fun loadFacilitiesWithQuery(): ObservableTransformer<LoadFacilitiesWithQuery, RegistrationFacilitySelectionEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .switchMap { effect ->
```

Comment:
Should we observe on `io` here or subscribing on `io` for the nested chain is fine?

Reply:
If we are invoking a reactive method, I generally prefer to do it in the nested chain.